### PR TITLE
Fix issue #52

### PR DESCRIPTION
**Pull request by Curzy**
#52 
This error caused by increasing index of bytes data
before getting value increase index, then overflow

So, i fixed with ordering sequence of function `decode_document` binary part

### CODE REVIEW

**Comment by Parkayun** [comment]
@Curzy Could you add some test cases for #52 ?

**Comment by Curzy** [comment]
@Parkayun Sure. i will add soon :smile:

**Comment by Parkayun** [comment]
I think `test_binary.py` should be a `test_bytes.py`

File: bson/codec.py [hunk 1]
```diff
@@ -292,8 +293,8 @@ def decode_document(data, base, as_array=False):
             base, value = decode_document(data, base, as_array=True)
         elif element_type == 0x05:  # binary
```
Comment (Curzy) [inline bson/codec.py]:
```
Line#143
 ELEMENT_TYPES = {
     0x01: "double",
     0x02: "string",
     0x03: "document",
     0x04: "array",
     0x05: "binary",
     0x07: "object_id",
     0x08: "boolean",
     0x09: "UTCdatetime",
     0x0A: "none",
     0x10: "int32",
     0x12: "int64"
 }
```

and this in-line comment,

I named test as binary

Comment (Curzy) [inline bson/codec.py]: So, change `test_binary.py` or `test_bytes.py` and change element types 

or

not to change?

Comment (Parkayun) [inline bson/codec.py]: Yes, you are right, I forgot the spec http://bsonspec.org/spec.html

**Comment by Parkayun** [comment]
Thanks 👍

**Pull request merged**